### DISCUSSION TITLE
SVG to PNG with transparency in php

### DISCUSSION THREAD

**Discussion by vlabob** [ImageMagick/ImageMagick#5593] [Help] [answered]
Hello guys,

I need to create PNG images out of SVG charts with transparent background.
I've been trying to make it work for some sdays now with no luck, so I wanted get help from first hand.
Here is my setup
![image](https://user-images.githubusercontent.com/19279453/192205719-61ac151c-f781-401e-93d8-d9bdae76c69e.png)

libPNG 1.5.13
Inkscape 0.92.2 (5c3e80d, 2017-08-06)

and here is code I'm using:

	$svg = '<?xml version="1.0"?>
			<svg version="1.1" x="0px" y="0px" viewBox="0 0 560 560" enable-background="new 0 0 280 560" xml:space="preserve">
				<polygon id="another_id" fill="green" stroke="black" stroke-width="5" points="200,100 131,5 19,41 19,159 131,195 "></polygon>
				<rect width="300" height="100" style="fill:rgb(0,0,255);stroke-width:3;stroke:rgb(0,0,0)" />
			</svg>';
	$image = new Imagick();
	$image->newImage(840, 560, new ImagickPixel('transparent'));
	$image->readImageBlob($svg);
	$image->setImageBackgroundColor(new ImagickPixel('transparent'));
	$image->setImageFormat("png64");
	$image->setOption('png:format', 'png64');
	
	$pngtmpFilePath = getcwd().'/render/test.png';
	$image->writeImage($pngtmpFilePath);

	echo $pngtmpFilePath;

I've searched for solutions on similar issues with now luck on solving it. Just need some way to debug it
and find where it breaks. Any help is most appreciated.

Thanks in advance.

**Comment by snibgo** [comment]
See the documentation (including User Contributed Notes) at https://www.php.net/manual/en/imagick.setbackgroundcolor.php

**Comment by vlabob** [comment]
I'm not sure why do you think that I didn't already looked at it.
Also that contribution is 10 years old and if I use that syntax it throws exception:
"Fatal error:  Uncaught ImagickException: Can not process empty Imagick object"

**Comment by snibgo** [comment]
Oh, okay. Sorry.

This is an ImageMagick forum, but your question is about IMagick. I suggest you ask on an IMagick forum.

**Comment by fmw42** [comment, marked as answer]
Do you need to make the white in the SVG into transparent? If so, then see transparentpaintimage() at https://www.php.net/manual/en/imagick.transparentpaintimage.php rather than setImageBackgroundColor()

**Comment by fmw42** [comment]
$svg = '<?xml version="1.0"?>
		<svg version="1.1" x="0px" y="0px" viewBox="0 0 560 560" enable-background="new 0 0 280 560" xml:space="preserve">
			<polygon id="another_id" fill="green" stroke="black" stroke-width="5" points="200,100 131,5 19,41 19,159 131,195 "></polygon>
			<rect width="300" height="100" style="fill:rgb(0,0,255);stroke-width:3;stroke:rgb(0,0,0)" />
		</svg>';
$image = new Imagick();
$image->readImageBlob($svg);
$imagick->transparentPaintImage("white", 0, 0, false);
$image->setImageFormat("png64");
$pngtmpFilePath = getcwd().'/render/test.png';
$image->writeImage($pngtmpFilePath);
echo $pngtmpFilePath;

If I save your svg data to a file, tmp.svg, then the above should be the equivalent of making the pure white into transparent as follows:

`convert tmp.svg -fuzz 0 -transparent white tmp.png
`

![tmp](https://user-images.githubusercontent.com/13458218/192388939-291ceb8e-5f04-4c28-9bc2-e4b9dd904983.png)

Increase the fuzz value to make near white pixels transparent.

**Comment by vlabob** [comment]
Thanks guys for answering in such short notice.
_**transparentPaintImage()**_ did the job.
I have some white details that need to stay white, but I can deal with that.
